plugin-core/src/main/java/appland/actions/QuickReviewAction.java

```diff
@@ -0,0 +1,267 @@
+package appland.actions;
+
+import appland.webviews.navie.NavieEditorProvider;
+import appland.webviews.navie.NaviePromptSuggestion;
+import com.intellij.icons.AllIcons;
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.actionSystem.ActionUpdateThread;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.popup.JBPopupFactory;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vfs.VirtualFile;
+import git4idea.commands.Git;
+import git4idea.commands.GitCommand;
+import git4idea.commands.GitLineHandler;
+import git4idea.repo.GitRepository;
+import git4idea.repo.GitRepositoryManager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+public class QuickReviewAction extends AnAction implements DumbAware {
+    public static final String ACTION_ID = "appmap.quickReview";
+    private static final String LAST_PICKED_REF_KEY = "appmap.quickReview.lastPickedRef";
+    private static final List<String> COMMON_MAIN_BRANCHES = Arrays.asList(
+            "main", "master", "develop", "release", "staging", "testing", "qa", "prod"
+    );
+
+    @Override
+    public @NotNull ActionUpdateThread getActionUpdateThread() {
+        return ActionUpdateThread.BGT;
+    }
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent e) {
+        var project = e.getProject();
+        if (project == null) {
+            return;
+        }
+
+        var repositoryManager = GitRepositoryManager.getInstance(project);
+        var repositories = repositoryManager.getRepositories();
+        if (repositories.isEmpty()) {
+            return;
+        }
+
+        // For now, we'll just use the first repository
+        var repository = repositories.get(0);
+
+        new Task.Backgroundable(project, "Fetching Git Refs", true) {
+            private List<GitRef> refs;
+            private boolean dirty = false;
+
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                try {
+                    refs = getItems(project, repository);
+                    dirty = isDirty();
+                } catch (VcsException ex) {
+                    throw new RuntimeException("Failed to fetch Git refs", ex);
```

Comment:
Would this error be displayed to the user?

If I'm not mistaken, implementing `onThrowable` would allow to show a notification or message.